lazy_tensor_core/lazy_tensor_core/csrc/LazyShapeDtype.cpp

```diff
@@ -43,14 +43,51 @@
  *
  */
 
+#include <functional>
 #include "lazy_tensor_core/csrc/ts_backend/LazyShapeDtype.h"
 #include "aten/src/ATen/WrapDimUtils.h"
 #include "torch/csrc/api/include/torch/enum.h"
-
 namespace torch_lazy_tensors {
 namespace ir {
 namespace ops {
 
+std::vector<std::vector<int64_t>> compute_shape_std(const at::Tensor & self, bool unbiased){
+  return compute_shape_std(self, c10::nullopt, c10::nullopt, false);
+}
+std::vector<std::vector<int64_t>> compute_shape_std(const at::Tensor & self, at::IntArrayRef dim, bool unbiased, bool keepdim){
+  return compute_shape_std(self, dim, c10::nullopt, keepdim);
+}
+std::vector<std::vector<int64_t>> compute_shape_std(const at::Tensor & self, c10::optional<at::IntArrayRef> dim, c10::optional<int64_t> correction, bool keepdim){
+  if (dim.has_value()) {
+    auto sizes = self.sizes().vec();
+    if (keepdim) {
+      for (auto reduce_dim:  dim.value()) {
+        sizes[at::maybe_wrap_dim(reduce_dim, self)] = 1;
+      }
+      return {sizes};
+    } else {
+      std::vector<int64_t> reduce_dims(dim.value().begin(), dim.value().end());
```

Comment:
@bdhirsh Mind take a look on this part? I'm not sure.

Reply:
oh okay - this one's more complicated because it uses `TensorIterator::make_reduction()` in core. Maybe that would be a good candidate to port to structured instead of trying to replicate the main logic here 😛 .

I'm happy to take a look / try to confirm if you have most of the logic. It's a little unfortunate that we don't have an automated way to check their correctness for now. PyTorch/XLA runs our device-agnostic tests in core on their backend - so once they start using these formulas then they'll get tested. Although I wonder if we can get the same testing for the `Lazy` devices sooner.